sql-to-dbsp-compiler/lib/json/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+#![deny(missing_docs)]
+
+//! Functions related to `JSON` support in DBSP
+
+/// The actual Json type.
+pub type Json = ijson::IValue;
+
+/// Parses the given string to [`Json`].
+///
+/// # Panics
+/// - Panics if it fails to deserialize to a valid Json.
+pub fn parse_json(value: impl AsRef<str>) -> Json {
+    serde_json::from_str(value.as_ref()).unwrap_or_else(|_| {
+        panic!(
+            "cannot deserialize given string to JSON: {}",
+            value.as_ref()
+        )
+    })
+}
+
+/// Returns `true` if the given string is a valid [`Json`].
+pub fn check_json(value: impl AsRef<str>) -> bool {
+    serde_json::from_str::<Json>(value.as_ref()).is_ok()
+}
+
+/// Extracts a `field` from the [`Json`] value.
+/// Use [`try_json_field`] for a non panicking implementation.
+///
+/// # Panics
+/// - Panics if the JSON value isn't a JSON object.
+/// - Panics if the `field` doesn't exist in this object.
+pub fn json_field(value: Json, field: impl AsRef<str>) -> Json {
+    value
+        .get(field.as_ref())
+        .expect("invalid: JSON_FIELD called with non existent field")
+        .clone()
+}
+
+/// Tries to extract a `field` from the [`Json`] value.
+pub fn try_json_field(value: Json, field: impl AsRef<str>) -> Option<Json> {
+    value.as_object()?.get(field.as_ref()).cloned()
+}
+
+/// Extracts the value at the given JSON index from this JSON array literal.
+/// Indexing starts from 1 to match other SQL functions.
+/// Use [`try_json_index`] for a non panicking implementation.
+///
+/// # Panics
+/// - Panics if the JSON value isn't a JSON array literal.
+/// - Panics if called with the index `0`.
+pub fn json_index(value: Json, mut idx: usize) -> Json {
+    idx = idx
+        .checked_sub(1)
```

Comment:
make this into a utility function which we can change easily if we change our minds